src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -342,6 +342,8 @@
     <!-- Non-null type is required for this node kind -->
     <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
     <Field Name="Operand" Type="BoundExpression"/>
+    <!-- Used by emit to determine whether this pointer indirection is by value or by ref -->
+    <Field Name="IsAssignmentAddress" Type="bool"/>
   </Node>
```

Comment:
I am not sure why do we need to calculate this value in Optimizer and limit that just to an assignment context. It feels like we can determine if a BoundPointerIndirectionOperator node is actually a dereference during an initial binding. If it is a target of a ```ref``` expression, it is not a dereference. If we capture this information during initial binding, the implementation wouldn't need to worry about other `ref` propagating scenarios like conditional expression, switch expression, etc.

Reply:
I like this refactoring, and went ahead and implemented it.